<a href="https://colab.research.google.com/github/apfoertn/Kaggle_Challenge/blob/master/Kaggle.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Project description

Goal of the project is to predict the price of a house based on 79 different explanatory variables.

Therefore, we are using the dataset from Dean De Cock

....

SalePrice is the targe variable --> most important


## Import Packages and Data

In [ ]:
#import all necessary packages

import pandas as pd 

In [ ]:
#import data from GitHub

train_data = pd.read_csv("https://raw.githubusercontent.com/apfoertn/Kaggle_Challenge/master/data/train.csv")
test_data = pd.read_csv("https://raw.githubusercontent.com/apfoertn/Kaggle_Challenge/master/data/test.csv")

## Having a first look

In [14]:
train_data.head(10)

Id  MSSubClass MSZoning  ...  SaleType  SaleCondition SalePrice
0   1          60       RL  ...        WD         Normal    208500
1   2          20       RL  ...        WD         Normal    181500
2   3          60       RL  ...        WD         Normal    223500
3   4          70       RL  ...        WD        Abnorml    140000
4   5          60       RL  ...        WD         Normal    250000
5   6          50       RL  ...        WD         Normal    143000
6   7          20       RL  ...        WD         Normal    307000
7   8          60       RL  ...        WD         Normal    200000
8   9          50       RM  ...        WD        Abnorml    129900
9  10         190       RL  ...        WD         Normal    118000

[10 rows x 81 columns]

In [17]:
train_data.dtypes

Id                 int64
MSSubClass         int64
MSZoning          object
LotFrontage      float64
LotArea            int64
                  ...   
MoSold             int64
YrSold             int64
SaleType          object
SaleCondition     object
SalePrice          int64
Length: 81, dtype: object

We have numerical and non-numerical columns

### Have a look at the targevt variable "SalePrice"

In [20]:
train_data.SalePrice.describe()

count      1460.000000
mean     180921.195890
std       79442.502883
min       34900.000000
25%      129975.000000
50%      163000.000000
75%      214000.000000
max      755000.000000
Name: SalePrice, dtype: float64



*   average sale price is around 180921 dollars
*   the cheapest house costs 34900 dollars
*   the most expensive house costs 755000 dollars






In [16]:
train_data.Alley.head(20)

0     NaN
1     NaN
2     NaN
3     NaN
4     NaN
5     NaN
6     NaN
7     NaN
8     NaN
9     NaN
10    NaN
11    NaN
12    NaN
13    NaN
14    NaN
15    NaN
16    NaN
17    NaN
18    NaN
19    NaN
Name: Alley, dtype: object

*   Already a lot of NaN Values, for example in column "Alley" 
*   need to clean data

## Data cleaning